Solutions/ImpervaCloudWAF/Parsers/ImpervaWAFCloud

```diff
@@ -0,0 +1,66 @@
+// Usage Instruction : 
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name and alias as ImpervaWAFCloud.
+// Function usually takes 10-15 minutes to activate. You can then use function alias from any other queries (e.g. JiraAudit | take 10).
+// Reference : Using functions in Azure monitor log queries : https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+ImpervaWAFCloud_CL 
+ | where Message has "CEF"
+ | extend logs = split(Message, "|") 
+ | extend product = logs[1]
+ | extend type = logs[2]
+ | extend severity = logs[5]
+ | extend msg = logs[7]
+ | extend  Properties = extract_all(@"(?P<key>[^=\s]+)?=(?P<value>(?:[\\]=|[^=])+)(?:\s|$)", dynamic(["key","value"]), tostring(msg))
+| mv-apply Properties on (
+    summarize make_bag(pack(tostring(Properties[0]), Properties[1]))
+)
+| evaluate bag_unpack(bag_) 
+| extend packed = pack(
+                    replace(@"\s","",cs1Label), cs1,
+                    replace(@"\s","",cs2Label), cs2,
+                    replace(@"\s","",cs3Label), cs3,
+                    replace(@"\s","",cs4Label), cs4,
+                    replace(@"\s","",cs5Label), cs5,
+                    replace(@"\s","",cs6Label), cs6,
+                    replace(@"\s","",cs7Label), cs7,
+                    replace(@"\s","",cs8Label), cs8
+                 )
+| evaluate bag_unpack(packed)
+| project-away logs, msg, cs1, cs1Label, cs2, cs2Label, cs3, cs3Label, cs4,cs4Label, cs5,cs5Label, cs6, cs6Label, cs7, cs7Label, cs8, cs8Label, Message
+| extend    EventVendor = "Imperva",
+            EventProduct = product,
+            EventType = type,
+            EventSeverity = severity,
+            DvcAction = act,
+            NetworkApplicationProtocol = app,
```

Comment:
The parser fails for below reasons,
1. There is an inbuilt function with same name as 'app' which cause ambiguity. Please refer below screenshot for the error
![image](https://user-images.githubusercontent.com/73389082/126327665-51aa9d8f-e51e-4f57-a7e8-3ba8e8d7bde8.png)

2. The sample data does not have values for **qstr**, **postbody** and **ver** fields

Reply:
1- The "app" field name is automatically parsed from predefined CEF log format, I can`t change this name. It is underlined but works as expected. 
2- I Added the samples with qstr, postbody and ver fields. Pls check again

Reply:
Thanks @averbn, the parser is working fine now